daffodil/predicate.py

```diff
@@ -47,8 +47,9 @@ def test_data_point(data_point):
             err_ret_val = getattr(test, "onerror", False)
             
             try: dp_val = data_point[key]
-            except KeyError: return err_ret_val 
-            
+            except KeyError: return err_ret_val
```

Comment:
thought of that, but as this is like _almost-never-happens-scenario_ i wanted to bypass that check each time